app/segments/drill/[id]/submission/input.js

```diff
@@ -402,14 +489,74 @@ export default function Input({ drillInfo, setToggleResult, setOutputData }) {
 
   //Function to handle "Next shot" button click
   const handleNextShotButtonClick = () => {
+    //close previous Banners
+    setEmptyInputBannerVisible(false);
+    setInvalidInputBannerVisible(false);
+
+    // Check if any inputs are empty
+    const inputValuesForShot = inputValues[displayedShot];
+    const isEmptyInput = Object.values(inputValuesForShot).some(
+      (value) => value === "",
+    );
+
     //Check if all inputs have been filled in
-    if (Object.keys(inputValues[displayedShot]).length === numInputs) {
+    if (Object.keys(inputValues[displayedShot]).length != numInputs) {
+      setEmptyInputBannerVisible(true);
+    } else if (isEmptyInput) {
+      setEmptyInputBannerVisible(true);
+    }
+    //check inputs are all numbers
+    else if (!validateInputs(inputValues[displayedShot])) {
+      setInvalidInputBannerVisible(true);
+    } else {
       setEmptyInputBannerVisible(false);
       setDisplayedShot(displayedShot + 1);
       setCurrentShot(currentShot + 1);
-    } else {
+    }
+  };
+
+  //Function to handle "Next shot" button click
+  const handleSubmitButtonClick = () => {
+    //close previous Banners
+    setEmptyInputBannerVisible(false);
```

Comment:
This entire part is a repeat of the above. Put it in a function or sth

Reply:
put it in another function because the end result was different but I can try to condense it into one

Reply:
I condensed both into "handleButton click, and have a check after errors to see if it the submit button